sklearn/cluster/_agglomerative.py

```diff
@@ -760,6 +760,19 @@ class AgglomerativeClustering(ClusterMixin, BaseEstimator):
         If "precomputed", a distance matrix (instead of a similarity matrix)
         is needed as input for the fit method.
 
+        .. deprecated:: 1.2
+            `affinity` was deprecated in version 1.2 and will be renamed to
+            `metric` in 1.4.
+
+    metric : str or callable, default='euclidean'
```

Comment:
I think that we should mention `None` for the moment.
We can document that if `None` then Euclidean distance will be computed.

```suggestion
    metric : str or callable, default=None
```